### PR TITLE
Always correct rs cua internal write errors

### PR DESCRIPTION
**Pull request by sandraros**
Symptom: issue #1807 error RS_CUA_INTERNAL_WRITE while pulling an old abapgit repository. Same issue as #562 which is solved by correcting manually the XML of the CUA interfaces (fastidious), or by regenerating the XML (but unfortunately sometimes the abapGit repository is from someone else).

Fix: Automatically fix this before RS_CUA_INTERNAL_WRITE is called. It's useless having this as a configuration option.

### CODE REVIEW

File: src/objects/zcl_abapgit_objects_program.clas.abap [hunk 1]
```diff
@@ -796,13 +807,22 @@ CLASS zcl_abapgit_objects_program IMPLEMENTATION.
...
 
+
+    IF is_cua-adm IS NOT INITIAL.
+      ASSIGN is_cua-adm TO <ls_adm>.
+    ELSE.
+      " issue #1807 automatic correction of CUA interfaces saved incorrectly in the past (ADM was not saved in the XML)
+      CLEAR ls_adm.
+      auto_correct_cua_adm( EXPORTING is_cua = is_cua CHANGING cs_adm = ls_adm ).
+      ASSIGN ls_adm TO <ls_adm>.
+    ENDIF.
```
Comment (larshp) [inline src/objects/zcl_abapgit_objects_program.clas.abap]: Idea was to reduce the amount of code in the method, suggest having

```abap
ls_adm = is_cua-adm.
auto_correct_cua_adm( EXPORTING is_cua = is_cua CHANGING cs_adm = ls_adm ).
```

intead of the 7 statements above

**Comment by kgorin8** [comment]
I think you should always autocorrect CUA. I just got a report from https://github.com/abapinho/abapinho_dev_tools that had ADM but CUA was still incorrect. Had to manually correct XMLs.

**Comment by sandraros** [comment]
@kgorin8 the repository contains 2 saplink files, not abapgit, what do you mean?

**Comment by kgorin8** [comment]
yes, i imported those and tried to transport it to another system with abapgit then. but auto_correct_cua_adm will return if ADM is X. which it was. and CUA will throw.

**Pull request merged**